rating-core/src/lib.rs

```diff
@@ -106,6 +106,30 @@ fn calc_q(performance: i32, attenuation: u32) -> f64 {
     performance as f64 - S * LOG_TABLE[attenuation as usize]
 }
 
+// calculate gp30 according to https://atcoder.jp/posts/170
+pub fn calc_gp30(rank: u32) -> u32 {
+    if rank > 30 {
+        0
+    } else if rank >= 15 {
+        16 - (rank - 15)
+    } else if rank >= 10 {
+        26 - 2 * (rank - 10)
+    } else {
+        match rank {
+            1 => 100,
+            2 => 75,
+            3 => 60,
+            4 => 50,
+            5 => 45,
+            6 => 40,
+            7 => 36,
+            8 => 32,
+            9 => 29,
+            _ => 0,
+        }
+    }
```

Comment:
完全に好みの問題なのですが、こちらの書き方でいかがでしょう？

```suggestion
    match rank {
        1 => 100,
        2 => 75,
        3 => 60,
        4 => 50,
        5 => 45,
        6 => 40,
        7 => 36,
        8 => 32,
        9 => 29,
        10..=14 => 26 - (rank - 10),
        15..=30 => 16 - (rank - 15),
        _ => 0,
    }
```

Reply:
え、Rustのmatchしゅごいになっています
勉強になります~!